pkg/downloader/manager.go

```diff
@@ -57,53 +57,75 @@ type Manager struct {
 	SkipUpdate bool
 	// Getter collection for the operation
 	Getters []getter.Prop
+	// Recursive mode collects the dependencies to the lowest level
+	Recursive bool
 }
 
 // Build rebuilds a local charts directory from a lockfile.
 //
 // If the lockfile is not present, this will run a Manager.Update()
 //
 // If SkipUpdate is set, this will not update the repository.
-func (m *Manager) Build() error {
+func (m *Manager) Build() (int, error) {
 	c, err := m.loadChartDir()
 	if err != nil {
-		return err
+		return 0, err
 	}
 
 	// If a lock file is found, run a build from that. Otherwise, just do
 	// an update.
 	lock, err := chartutil.LoadRequirementsLock(c)
 	if err != nil {
-		return m.Update()
+		return 0, m.Update()
 	}
 
 	// A lock must accompany a requirements.yaml file.
-	req, err := chartutil.LoadRequirements(c)
+	reqs, err := chartutil.LoadRequirements(c)
 	if err != nil {
-		return fmt.Errorf("requirements.yaml cannot be opened: %s", err)
+		return 0, fmt.Errorf("requirements.yaml cannot be opened: %s", err)
 	}
-	if sum, err := resolver.HashReq(req); err != nil || sum != lock.Digest {
-		return fmt.Errorf("requirements.lock is out of sync with requirements.yaml")
+	if sum, err := resolver.HashReq(reqs); err != nil || sum != lock.Digest {
+		return 0, fmt.Errorf("requirements.lock is out of sync with requirements.yaml")
 	}
 
 	// Check that all of the repos we're dependent on actually exist.
 	if err := m.hasAllRepos(lock.Dependencies); err != nil {
-		return err
+		return 0, err
 	}
 
 	if !m.SkipUpdate {
 		// For each repo in the file, update the cached copy of that repo
 		if err := m.UpdateRepositories(); err != nil {
-			return err
+			return 0, err
 		}
 	}
 
 	// Now we need to fetch every package here into charts/
 	if err := m.downloadAll(lock.Dependencies); err != nil {
-		return err
+		return 0, err
 	}
 
-	return nil
+	myDeps := len(reqs.Dependencies)
+	if m.Recursive {
+		depM := *m
+		for _, dep := range reqs.Dependencies {
+			depM.ChartPath = filepath.Join(m.ChartPath, "charts", dep.Name)
+			depM.SkipUpdate = true
+			fmt.Fprintln(m.Out, "Getting requirements recursively for:", depM.ChartPath)
+			depOfDeps, err := depM.Build()
+			if err != nil {
+				return 0, err
+			}
+			myDeps += depOfDeps
+			if myDeps > 100 {
```

Comment:
For now, we could just move this to a `var MaxDepsPerChart = 100` setting at the top of manager. I'm not sure it makes sense to make this user-configurable yet (since realistically users will hit the upper storage bound before they are likely to hit this).